diff --git a/esm/esm.mjs b/esm/esm.mjs
index 154a7171db37a40aca835d33426734c364166163..18d282be50bf7a1ed8236fa37e6de373e1295b19 100644
--- a/esm/esm.mjs
+++ b/esm/esm.mjs
@@ -62,11 +62,10 @@ export const resolve = async (specifier, context, nextResolve) => {
     }
     if (DEFAULT_EXTENSIONS.some((ext) => specifier.endsWith(ext))) {
         specifier = specifier.startsWith('file://') ? specifier : pathToFileURL(specifier).toString();
-        const newUrl = `${specifier}.mjs`;
-        TRANSFORM_MAP.set(newUrl, fileURLToPath(specifier));
+        TRANSFORM_MAP.set(specifier, fileURLToPath(specifier));
         return {
             shortCircuit: true,
-            url: new URL(newUrl).href,
+            url: specifier,
             format: 'module',
         };
     }
